File: python/examples/responses/basic.py
from mirascope import llm

model: llm.Model = llm.use_model("openai/gpt-4o")
response: llm.Response = model.call("What is the capital of France?")

# Access response content
print(response.pretty())  # Human-readable output
print(response.texts)  # List of Text objects
print(response.content)  # All content parts (Text, ToolCall, Thought)
print(response.messages)  # All of the messages in the response.

File: typescript/examples/responses/basic.ts
/**
 * Basic response usage example.
 *
 * Demonstrates accessing the response object and its text content.
 *
 * Run with: bun run example examples/responses/basic.ts
 */
import { llm } from "mirascope";

const model = llm.model("openai/gpt-4o-mini");
const response = await model.call("What is the capital of France?");

// Prints the textual content of the response
console.log(response.text());
// Paris is the capital of France.
